### PR TITLE
onboarding checklist copy

### PR DESCRIPTION
**Pull request by jeff-bruemmer**
Updates to onboarding copy.

### CODE REVIEW

File: frontend/src/metabase/home/components/Onboarding/Onboarding.tsx [hunk 1]
```diff
@@ -409,17 +409,17 @@ export const Onboarding = () => {
...
                           </ExternalLink>
                         ) : (
                           t`dashboard with tabs`
                         )
-                      } and add text cards.`}</li>
-                      <li>{jt`add ${(<b key="filters">{t`filters`}</b>)} to dashboards and connect them to fields on questions to narrow the results.`}</li>
-                      <li>{t`drill-through your dashboard and charts to see more detailed data underneath.`}</li>
+                      } and add text, link, and iframe cards.`}</li>
+                      <li>{jt`Add ${(<b key="filters">{t`filters`}</b>)} to dashboards and connect them to fields on questions to narrow the results.`}</li>
+                      <li>{t`Drill-through charts on your dashboard, or set up custom destinations on click.`}</li>
```
Comment (trinya) [inline frontend/src/metabase/home/components/Onboarding/Onboarding.tsx:422]: Can we leave a variation of the old text here? not sure what the custom destinations are? I felt that people might need to understand what is drill through, therefore "to see more detailed data underneath" was there.

File: frontend/src/metabase/home/components/Onboarding/Onboarding.tsx [hunk 2]
```diff
@@ -446,7 +446,7 @@ export const Onboarding = () => {
               data-testid="subscription-item"
             >
               <Accordion.Control icon={<Icon name="subscription" />}>
-                {t`Get dashboard updates by email`}
+                {t`Get dashboard results by email or Slack`}
```
Comment (trinya) [inline frontend/src/metabase/home/components/Onboarding/Onboarding.tsx:449]: "Results" is our internal terminology that I don't think matches the mental model of a new user - e.g. I have a dashboard, I want to get a fresh/updated dashboard every day. Dashboard is not something that produces results, rather an overview of the data?

**Review by trinya — CHANGES_REQUESTED** [review]
Left comments near some of the changes - hope it works.

File: frontend/src/metabase/home/components/Onboarding/Onboarding.tsx (hunk 2 above)
Comment (jeff-bruemmer) [reply frontend/src/metabase/home/components/Onboarding/Onboarding.tsx:449]: Interesting. I was thinking "updates" would mean when people _edited_ the dashboard. "Results" would refer to the data displayed on the dashboard. I'll change to "Subscribe to a dashboard by email or Slack."

**Review by nemanjaglumac — APPROVED** [review]

**Pull request merged**